rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -1259,11 +1260,25 @@ RMW_Connext_Subscriber::create(
     });
 
   DDS_TopicDescription * sub_topic = DDS_Topic_as_topicdescription(topic);
-
-  if (nullptr != cft_name) {
-    rmw_ret_t cft_rc =
-      rmw_connextdds_create_contentfilteredtopic(
-      ctx, dp, topic, cft_name, cft_filter, &cft_topic);
+  std::string cft_topic_name;
+  if (nullptr != cft_name ||
+    nullptr != subscriber_options->filter_expression)
```

Comment:
(partially quoting myself from [this comment in another PR](https://github.com/ros2/rmw_connextdds/pull/13#discussion_r601774772))

As you probably saw, this implementation defines a series of "feature flags" in `static_config.hpp` to be able to support multiple versions of ROS2 from the same branch. This is to avoid having to create branches for different versions, which we'll probably end up doing starting with Galactic, but would have been a pain to maintain up until now, while the RMW was in very active development.

For now, I'd like to continue to adhere with this pattern so that the `master` branch remains buildable with Foxy, so I suggest introducing a new flag in `static_config.hpp` to make sure field `filter_expression` is only accessed in the appropriate ROS2 version. Something like:

```cpp
#ifndef RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS
#define RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS \
  (RMW_CONNEXT_RELEASE > RMW_CONNEXT_RELEASE_FOXY)
#endif /* RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS */
```
Then wrap code that accesses `subscriber_options->filter_expression` (here, and below on line 1273) with `#if RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS`.

Reply:
Oh, actually I don't like this style.
I found the `RMW_CONNEXT_HAVE_OPTIONS_PUBSUB` before, which seems a little weird.
But I'll follow your suggestion. :)

Reply:
It seems you don't want to use some specific branches (eloquent, foxy, etc) to manage the code, but just use one branch(master) with multiple macros.

Reply:
> This is to avoid having to create branches for different versions, which we'll probably end up doing starting with Galactic

This CFT feature in `rmw` will not be merged in ROS2 Galactic (but might be H version), so I have no idea which way I need to do it?
```
#define RMW_CONNEXT_RELEASE_GALACTIC        50

#ifndef RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS
#define RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS \
  (RMW_CONNEXT_RELEASE > RMW_CONNEXT_RELEASE_GALACTIC)
#endif /* RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS */
```

or 

do nothing. (based on your comment)

Reply:
If the feature will not make it into Galactice, I would use `RMW_CONNEXT_RELEASE_ROLLING` for now, i.e.:
```cpp
#ifndef RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS
#define RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS \
    (RMW_CONNEXT_RELEASE >= RMW_CONNEXT_RELEASE_ROLLING)
#endif /* RMW_CONNEXT_HAVE_CONTENT_FILTERED_TOPICS */
```

I will review all feature flags, introduce `RMW_CONNEXT_RELEASE_GALACTIC`, and bump the value of `RMW_CONNEXT_RELEASE_ROLLING` after this feature is merged.

I know having macros is not the best, but it was the "lesser evil" as the code base was in very active development to allow supporting multiple version without having to deal with merging a lot between branches.

Once the package is included in a release, I think we will create a dedicated `galactic` branch for maintenance purposes (only bug fixes and critical patches will be merged). We will also be able to use that branch for releases pre-Galactic (along with the existing `dashing` branch), and at that point we can consider cleaning up the main development branch (`master`) to only support Rolling and future releases.

Reply:
thank you. updated

Reply:
great, thanks!